### PR TITLE
Create 30 seconds  list  and move things around

### PR DESCRIPTION
**Pull request by kirjs**

### CODE REVIEW

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-info/snippet-info.component.scss [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-    position: sticky;
     float: right;
     top: 40px;
     max-width: 50%;
     padding: 35px 35px 15px 35px;
-    background-color: #c8ced3;
-    border: 1px solid #a4b7c1;
     margin-left: 30px;
 
     li {
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-info/snippet-info.component.scss]: So, for now there is no 'li' in 'snippets-info' class, you can remove it)

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-info/snippet-info.component.ts [hunk 2]
```diff
@@ -1,19 +1,11 @@
...
 export class SnippetInfoComponent {
-
-  isInfoOnMotion = false;
-
-  @HostListener('window:scroll', ['$event'])
-  onScrollPosition() {
-    this.isInfoOnMotion = window.pageYOffset > 40;
-  }
-
   scrollUp() {
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-info/snippet-info.component.ts]: It's not used any more

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-modal/snippet-overview.component.ts [hunk 3]
```diff
@@ -17,13 +17,9 @@ function getTagsStringList(tagsArray: Array<string>): string {
 function getSnippet(value): string {
   return `
 ---
-
 title: ${value.title}
-
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-modal/snippet-overview.component.ts]: You'd better to leave this spaces. Or it looks like this: 
image

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-modal/snippet-overview.component.ts [hunk 4]
```diff
@@ -17,13 +17,9 @@ function getTagsStringList(tagsArray: Array<string>): string {
...
-author: ${value.author ? `${value.author}` : `*your github username will be added*`}
-
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-modal/snippet-overview.component.ts]: The same as above

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-modal/snippet-overview.component.ts [hunk 5]
```diff
@@ -17,13 +17,9 @@ function getTagsStringList(tagsArray: Array<string>): string {
...
+author: ${value.author || `*your github username will be added*`}
 level: ${value.level}
-
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/create-snippet/snippet-modal/snippet-overview.component.ts]: +1

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-demo/snippet-demo.component.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { Component, Input } from '@angular/core';
+import { angularSampleCode } from '../shared';
+
+@Component({
+  selector: 'slides-snippet-demo',
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-demo/snippet-demo.component.ts]: codelab prefix?

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-list/snippet-list.component.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<button routerLink="new">New</button>
+<div *ngFor="let snippet of snippets; let i = index">
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-list/snippet-list.component.html]: How do you think about move it to "...app/angular-thirty-seconds/snippet-demo/snippet-demo.component.html" to try to get rid of "i !== demoIndex"?

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-list/snippet-list.module.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import { CommonModule } from '@angular/common';
+import { SnippetListComponent } from './snippet-list.component';
+import { SnippetDemoComponent } from '../snippet-demo/snippet-demo.component';
+import { CodeDemoModule } from '@codelab/code-demos';
+import { FormsModule } from '@angular/forms';
+import { RouterModule } from '@angular/router';
+
+@NgModule({
+  declarations: [SnippetListComponent, SnippetDemoComponent],
+  exports: [SnippetListComponent],
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-list/snippet-list.module.ts]: Do you really need to export this component?

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/angular-thirty-seconds-routing.module.ts [hunk 9]
```diff
@@ -1,20 +1,40 @@
...
+})
+class SnippetsResolver implements Resolve<Observable<any>> {
+  resolve() {
+    return from(fetch('https://gitcdn.link/repo/nycJSorg/30-seconds-of-angular/master/data/data.json')
+      .then(data => data.json())
+    );
+  }
+}
 
 const routes = RouterModule.forChild(
```
Comment (alex-bu-93) [inline apps/angular-thirty-seconds/src/app/angular-thirty-seconds/angular-thirty-seconds-routing.module.ts]:
```suggestion
const routes = [
  {path: 'new', component: CreateSnippetComponent},
  {path: '', component: SnippetListComponent, resolve: {snippets: SnippetsResolver}},
  {path: 'list', component: SnippetListComponent, resolve: {snippets: SnippetsResolver}},
  {path: ':id', component: SnippetComponent, resolve: {snippets: SnippetsResolver}},
];

@NgModule({
  imports: [
    RouterModule.forChild(routes),
    SnippetListModule,
  ],
  exports: [RouterModule]
})
export class AngularThirtySecondsRoutingModule {
}
```

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-list/snippet-list.module.ts (hunk 8 above)
Comment (kirjs) [reply apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-list/snippet-list.module.ts]: Yes

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-list/snippet-list.component.html (hunk 7 above)
Comment (kirjs) [reply apps/angular-thirty-seconds/src/app/angular-thirty-seconds/snippet-list/snippet-list.component.html]: Don't understand

File: apps/angular-thirty-seconds/src/app/angular-thirty-seconds/angular-thirty-seconds-routing.module.ts (hunk 9 above)
Comment (kirjs) [reply apps/angular-thirty-seconds/src/app/angular-thirty-seconds/angular-thirty-seconds-routing.module.ts]: But why?

**Pull request merged**